src/MatrixFields/MatrixFields.jl

```diff
@@ -78,7 +78,10 @@ export DiagonalMatrixRow,
     QuaddiagonalMatrixRow,
     PentadiagonalMatrixRow
 export FieldVectorKeys, FieldMatrixKeys, FieldVectorView, FieldMatrix
-export FieldMatrixWithSolver, ⋅
+export FieldMatrixWithSolver
+
+export ⋅
+const ⋅ = * # TODO: Remove this in the next major release of ClimaCore.
```

Comment:
This seems very dangerous to me. I do not want ClimaCore to export this symbol, and make it an alias for `*`.